weblate/static/loader-codemirror.js

```diff
@@ -0,0 +1,80 @@
+(function (CodeMirror) {
+    var userMentionList = [];
+    var currentRequest = null;
+
+    function getUserList(usernameSearch) {
+        currentRequest = $.ajax({
+            type: 'GET',
+            url: `/api/users/?username=${usernameSearch}`,
+            dataType: 'json',
+            beforeSend : function() {
+                if (currentRequest !== null) {
+                    currentRequest.abort();
+                }
+            },
+            success: function (data) {
+                userMentionList = data.results.map(function(user) {
+                    return {
+                        text: user.username,
+                        displayText: `${user.full_name} (${user.username})`
+                    }
+                });
+            },
+            error: function (jqXHR, textStatus, errorThrown) {
+                console.error(errorThrown);
+            }
+        });
+    }
+
+    CodeMirror.registerHelper('hint', 'userSuggestions', function (editor) {
+        var cur = editor.getCursor();
+        var curLine = editor.getLine(cur.line);
+
+        var end = cur.ch;
+        var start = curLine.lastIndexOf('@') + 1;
+        // Extract the current word from the current line using 'start' / 'end' value pair
+        var curWord = start !== end && curLine.slice(start, end);
+
+        if(curWord && curWord.length > 2) {
+            // If there is current word set, We can filter out users from the
+            // main list and display them
+            getUserList(curWord);
+        }
+
+        return { list: userMentionList, from: CodeMirror.Pos(cur.line, start), to: CodeMirror.Pos(cur.line, end) };
+    });
```

Comment:
This still looks wrong. What it should do instead is to register the helper with callback and use the async mode available in the CodeMirror hints:

> It is possible to set the async property on a hinting function to true, in which case it will be called with arguments (cm, callback, ?options), and the completion interface will only be popped up when the hinting function calls the callback, passing it the object holding the completions.